docs/aks/mdmonboarding/mdm_onboarding.ps1

```diff
@@ -0,0 +1,208 @@
+			    Stop-Transcript
+			    exit
+		    }
+		    try {
+			    Import-Module Az.Aks -ErrorAction Stop
+		    } catch {
+			    Write-Host("Could not import Az.Aks... Please reinstall this Module") -ForegroundColor Red
+			    Stop-Transcript
+			    exit
+		    }         
+	
+	    }
+	    2 { 
+		    Write-Host("")
+		    Stop-Transcript
+		    exit
+	    }
+    }
+}
+
+try {
+	Write-Host("")
+	Write-Host("Trying to get the current Az login context...")
+	$account = Get-AzContext -ErrorAction Stop
+	Write-Host("Successfully fetched current AzContext context...") -ForegroundColor Green
+	Write-Host("")
+} catch {
+	Write-Host("")
+	Write-Host("Could not fetch AzContext..." ) -ForegroundColor Red
+	Write-Host("")
+}
+
+
+if ($account.Account -eq $null) {
+	try {
+		Write-Host("Please login...")
+		Connect-AzAccount -subscriptionid $SubscriptionId
+	} catch {
+		Write-Host("")
+		Write-Host("Could not select subscription with ID : " + $SubscriptionId + ". Please make sure the ID you entered is correct and you have access to the cluster" ) -ForegroundColor Red
+		Write-Host("")
+		Stop-Transcript
+		exit
+	}
+} else {
+	if ($account.Subscription.Id -eq $SubscriptionId) {
+		Write-Host("Subscription: $SubscriptionId is already selected. Account details: ")
+		$account
+	} else {
+		try {
+			Write-Host("Current Subscription:")
+			$account
+			Write-Host("Changing to subscription: $SubscriptionId")
+			Set-AzContext -SubscriptionId $SubscriptionId
+		} catch {
+			Write-Host("")
+			Write-Host("Could not select subscription with ID : " + $SubscriptionId + ". Please make sure the ID you entered is correct and you have access to the cluster" ) -ForegroundColor Red
+			Write-Host("")
+			Stop-Transcript
+			exit
+		}
+	}
+}
+
+#
+#   Check AKS cluster existance and access check
+#
+Write-Host("Checking aks cluster exists...")
+$cluster = Get-AzAks -ResourceGroupName $ResourceGroupName -Name $clusterName  -ErrorVariable notPresent -ErrorAction SilentlyContinue
+if ($notPresent) {
+	Write-Host("")
+	Write-Host("Could not find Aks cluster. Please make sure that specified cluster exists: '" + $clusterName + "'is correct and you have access to the cluster") -ForegroundColor Red
+	Write-Host("")
+	Stop-Transcript
+	exit
+}
+Write-Host("Successfully checked specified cluster exists details...") -ForegroundColor Green
+
+$servicePrincipalClientId = $cluster.ServicePrincipalProfile.clientId
+$clusterResourceId = $cluster.Id
+
+#
+#  Add Monitoring Metrics Publisher role assignment to the AKS cluster resource
+#
+
+New-AzRoleAssignment -ApplicationId $servicePrincipalClientId -scope $clusterResourceId -RoleDefinitionName "Monitoring Metrics Publisher"  -ErrorVariable assignmentError -ErrorAction SilentlyContinue
+
+if ($assignmentError) {
+
+    if($assignmentError.Exception -match "role assignment already exists") {
```

Comment:
New-AzRoleAssignment always returns the text in English? checking the code might be better.

Reply:
Not sure how the rest API handles localization. For now, I verified whether role assignment exists or not through GET call.